presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryBatchStreamReader.java

```diff
@@ -184,16 +183,11 @@ private void openRowGroup()
     {
         // read the dictionary
         if (!dictionaryOpen && dictionarySize > 0) {
-            if (dictionary.length < dictionarySize) {
-                dictionary = new long[dictionarySize];
+            int previousLength = dictionary.length;
+            dictionary = longDictionaryProvider.getDictionary(streamDescriptor, dictionarySize);
+            if (previousLength < dictionarySize) {
                 systemMemoryContext.setBytes(sizeOf(dictionary));
```

Comment:
previously the systemMemoryContext was unconditionally set, when the dictionary has values, now we are accounting only when the dictionary grows.

Reply:
This is not truth. Previously memory is also only set when dictionary buffer grows.